game/src/Credits/Credits.gd

```diff
@@ -32,72 +32,85 @@ var label_settings_personel : LabelSettings
 @export
 var credits_list: VBoxContainer
 
+const title_str : String = "TITLE"
+
 # REQUIREMENTS:
 # * 1.5 Credits Menu
 # * SS-17
 
 # REQUIREMENTS
 # * FS-4
-func load_credit_file(path):
-	var roles = {}
+func _load_credit_file(path : String) -> Dictionary:
+	var roles := {}
 	var core_credits = FileAccess.open(path, FileAccess.READ)
+	if core_credits == null:
+		push_error("Failed to open credits file %s (error code %d)" % [path, FileAccess.get_open_error()])
+		return roles
 
 	while not core_credits.eof_reached():
-		var line = core_credits.get_csv_line()
+		var line := core_credits.get_csv_line()
+		var role := line[0].strip_edges().to_upper()
 
 		# If the line does not have an identifiable role or is empty then skip it
-		if line[0].is_empty():
+		if role.is_empty() or line.size() < 2:
+			if not (role.is_empty() and line.size() < 2):
+				push_warning("Incorrectly formatted credit line %s in %s" % [line, path])
 			continue
 
-		var role := line[0].to_upper()
-		var person : String = line[1] # Cannot infer with := as line[1] does not have a set type
+		var person := line[1].strip_edges()
 
-		if role not in roles.keys():
-			roles[role] = []
-			roles[role].push_back(person)
-		else:
-			roles[role].push_back(person)
+		if person.is_empty():
+			push_warning("Incorrectly formatted credit line %s in %s" % [line, path])
+			continue
+		if line.size() > 2:
+			push_warning("Extra entries ignored in credit line %s in %s" % [line, path])
 
+		if role not in roles:
+			roles[role] = [person]
+		else:
+			if person in roles[role]:
+				push_warning("Duplicate person %s for role %s in %s" % [person, role, path])
+			else:
+				roles[role].push_back(person)
+	if title_str in roles:
+		if roles[title_str].size() > 1:
+			push_warning("More than one %s: %s in %s" % [title_str, roles[title_str], path])
+			roles[title_str] = [roles[title_str][0]]
+	else:
+		push_warning("Credits file %s missing %s" % [path, title_str])
+	for role_list in roles.values():
+		role_list.sort_custom(func(a : String, b : String) -> bool: return a.naturalnocasecmp_to(b) < 0)
 	return roles
 
+func _add_label(node : Node, text : String, settings : LabelSettings) -> void:
+	var label := Label.new()
+	label.name = 'Label' + text
+	label.text = text
+	label.horizontal_alignment = HORIZONTAL_ALIGNMENT_CENTER
+	label.label_settings = settings
+	node.add_child(label)
+
 # REQUIREMENTS:
 # * UI-34, UI-35
-func make_project_credits(project):
+func _add_project_credits(project) -> void:
```

Comment:
Why are you adding a underscore prefix to the identifier? Afaik this should only apply to the methods that are related to signals and other such godot engine related methods. The [gd script style guide](https://docs.godotengine.org/en/stable/tutorials/scripting/gdscript/gdscript_styleguide.html) suggests that methods should just be snake case.

It also has no whitespace between the identifier and colon for static typing

```suggestion
func add_project_credits(project) -> void:
```

Reply:
> Prepend a single underscore (_) to virtual methods functions the user must override, private functions, and private variables

I was using the underscore in the private function sense here, mainly so I could assume the Dictionary argument is formatted correctly (no empty list or non-list values).

Reply:
ah ok, makes sense. Then ignore the rest of my comments on here

Reply:
No worries, in practice private functions are often connected to signals. Although the underscore doesn't actually prevent functions and variables from being accessed externally, it does stop them from appearing in the list of autocomplete options in the Godot editor, and generally makes it clear that you shouldn't be using/accessing the function/variable unless you know what you're doing.